liblouis/lou_translateString.c

```diff
@@ -3679,71 +3679,111 @@ translateString(const TranslationTableHeader *table, int mode, int currentPass,
 	return 1;
 } /* first pass translation completed */
 
+static int
+isHyphen(const TranslationTableHeader *table, widechar c) {
+	TranslationTableRule *rule;
+	TranslationTableOffset offset = findCharOrDots(c, 0, table)->otherRules;
+	while (offset) {
+		rule = (TranslationTableRule *)&table->ruleArea[offset];
+		if (rule->opcode == CTO_Hyphen) return 1;
+		offset = rule->dotsnext;
+	}
+	return 0;
+}
+
+/**
+ * Hyphenate an input string which can either be text (mode = 0) or braille (mode = 1). If
+ * the input is braille, back-translation will be performed with `tableList'. The input
+ * string can contain any character (even space), but only break points within words
+ * (between letters) are considered. If the string can not be broken before the character
+ * at index k, the value of `hyphens[k]' is '0'. If it can be broken by inserting a hyphen
+ * at the break point, the value is '1'. If it can be broken without adding a hyphen, the
+ * value is '2'.
+ */
 int EXPORT_CALL
 lou_hyphenate(const char *tableList, const widechar *inbuf, int inlen, char *hyphens,
 		int mode) {
 #define HYPHSTRING 100
 	const TranslationTableHeader *table;
-	widechar workingBuffer[HYPHSTRING];
-	int k, kk;
+	widechar textBuffer[HYPHSTRING];
+	char *textHyphens;
+	int *inputPos;
+	int k;
+	int textLen;
 	int wordStart;
-	int wordEnd;
 	table = lou_getTable(tableList);
 	if (table == NULL || inbuf == NULL || hyphens == NULL ||
 			table->hyphenStatesArray == 0 || inlen >= HYPHSTRING)
 		return 0;
 	if (mode != 0) {
-		k = inlen;
-		kk = HYPHSTRING;
-		if (!lou_backTranslate(tableList, inbuf, &k, &workingBuffer[0], &kk, NULL, NULL,
-					NULL, NULL, NULL, 0))
+		int brailleLen = inlen;
+		textLen = HYPHSTRING;
+		inputPos = malloc(textLen * sizeof(char));
```

Comment:
Made a mistake here. Must be `sizeof(int)`. And I have a few other fixes coming up.